src/tools/crossgen2/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -202,9 +202,10 @@ private void CompileMethodInternal(IMethodNode methodCodeNodeNeedingCode, Method
 #if SUPPORT_JIT
                         _lastException.Throw();
 #else
-                        if (_lastException.SourceException is TypeSystemException)
+                        if (_lastException.SourceException is TypeSystemException || _lastException.SourceException is BadImageFormatException)
```

Comment:
Where does the BadImageFormatException come from?

There's a lot of `throw new BadImageFormatException` in the type system codebase that should really be `ThrowHelper.ThrowBadImageFormatException` - if it's one of those, we should just fix that and it will fall out nicely everywhere.

Reply:
I removed all my catches and no longer even see `BadImageFormatException` getting thrown so it looks like the linker bug got fixed. Easy!